tests/cloner_test.go

```diff
@@ -574,6 +575,75 @@ var _ = Describe("all clone tests", func() {
+					// the value is simply extracted from the original PVC's spec.
+					if cloneType == "network" && sourceVolumeMode == v1.PersistentVolumeFilesystem {
+						Skip("Clone strategy and volume mode combination requires of size-detection pod")
+					}
+
+					if sourceVolumeMode != targetVolumeMode && cloneType == "csivolumeclone" {
+						Skip("csivolumeclone only works for the same volumeMode")
+					}
+
+					if (sourceVolumeMode == v1.PersistentVolumeBlock || targetVolumeMode == v1.PersistentVolumeBlock) && !f.IsBlockVolumeStorageClassAvailable() {
+						Skip("Storage Class for block volume is not available")
+					}
+
+					dataVolume := utils.NewDataVolumeWithHTTPImportAndStorageSpec(dataVolumeName, "200Mi", fmt.Sprintf(utils.TinyCoreIsoURL, f.CdiInstallNs))
+					dataVolume.Spec.Storage.VolumeMode = &sourceVolumeMode
+					dataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, dataVolume)
+					Expect(err).ToNot(HaveOccurred())
+					f.ForceBindPvcIfDvIsWaitForFirstConsumer(dataVolume)
+					sourcePvc, err := f.K8sClient.CoreV1().PersistentVolumeClaims(dataVolume.Namespace).Get(context.TODO(), dataVolume.Name, metav1.GetOptions{})
+					Expect(err).ToNot(HaveOccurred())
+
+					// We attempt to create the sizeless DV
+					targetDV := utils.NewDataVolumeForCloningWithEmptySize("target-dv", sourcePvc.Namespace, sourcePvc.Name, nil, &targetVolumeMode)
+					targetDataVolume, err := utils.CreateDataVolumeFromDefinition(f.CdiClient, f.Namespace.Name, targetDV)
+					Expect(err).ToNot(HaveOccurred())
+					targetPvc, err := utils.WaitForPVC(f.K8sClient, targetDataVolume.Namespace, targetDataVolume.Name)
+					Expect(err).ToNot(HaveOccurred())
+					f.ForceBindPvcIfDvIsWaitForFirstConsumer(targetDataVolume)
+
+					By("Wait for target PVC Bound phase")
+					err = utils.WaitForPersistentVolumeClaimPhase(f.K8sClient, f.Namespace.Name, v1.ClaimBound, targetPvc.Name)
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Wait for target DV Succeeded phase")
+					err = utils.WaitForDataVolumePhaseWithTimeout(f.CdiClient, f.Namespace.Name, cdiv1.Succeeded, "target-dv", cloneCompleteTimeout)
+					Expect(err).ToNot(HaveOccurred())
+
+					diskImagePath := filepath.Join(testBaseDir, testFile)
+
+					By("Source file system pvc md5summing")
+					sourceMD5, err := f.GetMD5(f.Namespace, sourcePvc, diskImagePath, crossVolumeModeCloneMD5NumBytes)
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Deleting verifier pod")
+					err = utils.DeleteVerifierPod(f.K8sClient, f.Namespace.Name)
+					Expect(err).ToNot(HaveOccurred())
+					_, err = utils.WaitPodDeleted(f.K8sClient, utils.VerifierPodName, f.Namespace.Name, verifyPodDeletedTimeout)
+					Expect(err).ToNot(HaveOccurred())
+
+					By("Target file system pvc md5summing")
+					targetMD5, err := f.GetMD5(f.Namespace, targetPvc, diskImagePath, crossVolumeModeCloneMD5NumBytes)
+					Expect(err).ToNot(HaveOccurred())
+					Expect(sourceMD5 == targetMD5).To(BeTrue())
```

Comment:
I think
```go
Expect(sourceMD5).To(Equal(targetMD5))
```
is easier to read. Also I would check after I already delete the verifier pod. If the comparison fails in this order, we won't delete the verifier pod. (Note it is currently not a problem since the namespace will get deleted, but better cleanup)